app/src/main/java/com/willowtree/vocable/utils/FaceTrackingManager.kt

```diff
@@ -51,11 +48,20 @@ class FaceTrackingManager(
         activity.windowManager.defaultDisplay.getMetrics(displayMetrics)
 
         if (BuildConfig.USE_HEAD_TRACKING && checkIsSupportedDevice()) {
+
+            cameraPermissionsHelper.permissionCallback = { isGranted ->
+                if (isGranted) {
+                    faceTrackingPermissions.enableFaceTracking()
+                } else {
+                    faceTrackingPermissions.disableFaceTracking()
+                }
+            }
+
             CoroutineScope(Dispatchers.Main).launch {
```

Comment:
I think I missed this in a previous PR, but this is dangerous since there is no bound to how long we're running this block. Could result in trying to hide and show things that aren't there if the owner of this instance (Activity/Fragment) is gone. 

My recommendation is to make `initialize()` suspend, and then whoever is calling initialize is running that coroutine within their own lifecycle-bound scope.

Relating it to the current work: I'm actually wondering if you can simplify the back-and-forth of having the permission callback, and then updating faceTrackingPermissions, and then observing that update to faceTrackingPermissions. Would abstracting over the callbacks with a coroutine be possible here to make the code read more synchronously?

Reply:
Re: our discussion offline, the dream of making permissions request work with suspend functions looks like an impossibility for now. Did some refactoring to move CameraPermissionsHelper func into FaceTrackingPermissions since they do the same thing, which eliminated the need for the callbacks.